src/js/engagement_view/src/components/graphDisplay/GraphDisplay.tsx

```diff
@@ -39,36 +32,42 @@ const defaultClickedState = (): ClickedNodeState => {
     return null;
 };
 
+async function updateGraphAndSetState(
+    lensName: any,
+    state: any,
+    setState: any
+) {
+    if (lensName) {
+        await updateGraph(lensName, state as GraphState, setState); // state is safe cast, check that lens name is not null
+
+        console.log(
+            "setState",
+            await updateGraph(lensName, state as GraphState, setState)
+        );
+    }
+}
+
 const GraphDisplay = ({ lensName, setCurNode }: GraphDisplayProps) => {
     const fgRef: any = useRef(); // fix graph to canvas
     const [state, setState] = useState(defaultGraphDisplayState(lensName));
-
-    async function updateGraphAndSetState() {
-        if (lensName) {
-            await updateGraph(lensName, state as GraphState, setState); // state is safe cast, check that lens name is not null
-        }
-    }
+    const [clickedNode, setClickedNode] = useState(defaultClickedState());
+    const [highlightNodes, setHighlightNodes] = useState(new Set());
+    const [highlightLinks, setHighlightLinks] = useState(new Set());
+    const [hoverNode, setHoverNode] = useState(null);
```

Comment:
i also worry here that hoverNode, setHoverNode are untyped since we don't know what the `null` could be. I tihnk you can do like

```
hoverNodeBackingModel: GoodType | null = null; 
const [hoverNode, setHoverNode] = useState(hoverNodeBackingModel);
```

Reply:
in this case, the null in useState is the initial value for the graph, the hoverNodes are typed elsewhere, so this should be okay, but thak you for being so thoughtful here.